common/document_parser/lib/responsibility_parse/responsibility_parser.py

```diff
@@ -316,14 +364,50 @@ def parse_responsibility_section(self, resp_section):
             # or a new line that should be captured as a new line
             else:
                 if numbering:
+                    # in some cases, there are junk sections before (inside the overall responsibiltiy section) which have
+                    # matched the logic above and acted like responsibilties, however we want to clear out the cache and start
+                    # assigning responsibilities from the current line
+                    if line_text.upper().strip()=="RESPONSIBILITIES":
+                        section_resp_lines = []
+                        new_role_start_metadata = {
+                            "n_periods": 0,
+                            "n_parenthesis": 0,
+                            "n_numbers": 0,
+                            "n_letters": 0,
+                        }
+                        next_numbering_is_role = True
+                        continue
+
+                    # example of this edge case (the l. acts as a new numbering)
+                    # (9) Conduct inquiries, inspections, and investigations as directed by
+                    # the DJS or CJCS using the procedures and guidance IAW references a through
+                    # l.
+                    # (10) Provide assistance to the CCMD IG offices as requested.
+                    #
+                    # if there is no "text" for the line, and the next line has numbering, this is a continuation of the last
+                    # line
+                    if not line_text.strip():
+                        try:
+                            next_line_numbering, next_line_text = self.extract_numbering(resp_section[i+1])
+                            if next_line_numbering:
+                                section_resp_lines[-1] += f" {resp_line}"
+                                continue
+                        except:
+                            section_resp_lines[-1] += f" {resp_line}"
+                            continue
 
                     # A new role is being assigned a set of responsibilities, so capture this as a new grouping
-                    if multiple_roles_present and \
-                            self.is_larger_numbering(current_role_numbering,numbering) and \
```

Comment:
It looks like `current_role_numbering` isn't used anymore since these lines were removed, but it's still assigned in a few places